source/logHandler.py

```diff
@@ -31,22 +31,33 @@
 CO_E_OBJNOTCONNECTED = -2147220995
 EVENT_E_ALL_SUBSCRIBERS_FAILED = -2147220991
 LOAD_WITH_ALTERED_SEARCH_PATH=0x8
+_NVDA_CODE_PATH = os.path.dirname(__file__)
+"""Store path in which NVDA code is placed.
+We cannot use `globalVars.appDir`, since for binary builds it points to the directory with NVDA binaries,
+whereas for compiled versions NVDA's code files are in `library.zip`.
+"""
 
 
 def isPathExternalToNVDA(path: str) -> bool:
 	""" Checks if the given path is external to NVDA (I.e. not pointing to built-in code). """
 	if(
 		path[0] != "<"
 		and os.path.isabs(path)
-		and not os.path.normpath(path).startswith(sys.path[0] + "\\")
+		and not os.path.normpath(path).startswith(_NVDA_CODE_PATH + "\\")
+		or(
+			globalVars.appArgs.configPath is not None  # Handle messages logged before config is initialized
+			and path.startswith(globalVars.appArgs.configPath)
+		)
 	):
 		# This module is external because:
 		# the code comes from a file (fn doesn't begin with "<");
-		# it has an absolute file path (code bundled in binary builds reports relative paths); and
-		# it is not part of NVDA's Python code (not beneath sys.path[0]).
+		# it has an absolute file path (code bundled in binary builds reports relative paths);
+		# it is not part of NVDA's Python code; and
+		# it is placed in NVDA's config, so belongs to an add-on or a plugin in the scratchpad.

```

Comment:
should this be "or" not "and"?

```suggestion
		# it has an absolute file path (code bundled in binary builds reports relative paths); and
		# it is not part of NVDA's Python code
		# (i.e. outside of NVDA directory or in NVDA's config, so belongs to an add-on or a plugin in the scratchpad).
```